### PR TITLE
Factory setting mode

### PR DESCRIPTION
**Pull request by Peguy-WANDA**

### CODE REVIEW

**Review by dmohns — COMMENTED** [review]
I think I understand what this is trying to do. However, I think the user options are starting to get really complicated. I.e. how is the user supposed to interact with the Keypad? 

Could you please create a documentation `user-manual.md` in the `docs` folder that explains all the options and flows for the user?

File: firmware/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.pio
+.vscode/.browse.c_cpp.db*
+.vscode/c_cpp_properties.json
+.vscode/launch.json
+.vscode/ipch
```
Comment (dmohns) [inline firmware/.gitignore:5]: Hey @Peguy-WANDA you seem to have this file in every PR you send 😉 Please, either delete it locally, or commit file-by-file and not all files.

Comment (Peguy-WANDA) [reply firmware/.gitignore:5]: Hello Daniel, Ok it was a mistake sorry, So let me remove it

**Comment by frimic-solarly** [comment]
> I think I understand what this is trying to do. However, I think the user options are starting to get really complicated. I.e. how is the user supposed to interact with the Keypad?
> 
> Could you please create a documentation `user-manual.md` in the `docs` folder that explains all the options and flows for the user? Or maybe that's something @frimic-solarly can help out with?

Hello Daniel, I think, if I'm not mistaken, that the whole factory settings menu is available for technicians but not for the end-user. @Peguy-WANDA will start the testing and will do videos and pictures for us to better understand how this will be articulated.
I'll also take care of the documentation to make it clear for any newcomers.

**Review by dmohns — APPROVED** [review]

**Pull request merged**